### ISSUE TITLE
iOS app crashing after 0.2.2 Upgrade

### ISSUE THREAD

**Issue by connorpmullins** [AzizAK/react-native-detector#36]
Apologies in advance - this is a less-helpful issue report than I would like to make.

After upgrading `react-native-detector` to v0.2.2, our iOS app is crashing on launch with the error `Invariant Violation: Native module cannot be null.`

After downgrading back to v0.2.1, the issue was immediately resolved.

It's not pressing for us to get this fixed, and I don't expect help without me providing more information. I just want to open this issue as a central place for others to report on it if they have the same problem.

**Comment by aamikus** [comment]
I see that this is due to the change in the `NativeEventEmitter` initialization.
https://github.com/AzizAK/react-native-detector/compare/v0.2.1...v0.2.2#diff-0b5adbfe7b36e4ae2f479291e20152e33e940f7f265162d77f40f6bdb5da7405L9

From the react-native docs the first argument of the `NativeEventEmitter` constructor is required on iOS:
```
Params:
nativeModule – the NativeModule implementation. This is required on IOS and will throw an invariant error if undefined.
```

So I am curious why this change was made.

**Comment by qzaid** [comment]
Yeah I am having the same issue

**Comment by Ahmedhamed77** [comment]
just started to use the package and got the error ```new NativeEventEmitter()` requires a non-null argument```, any updates on that ?

**Comment by feraswfares** [comment]
same here +1

**Comment by AzizAK** [comment]
Issue has been fixed #41 release 0.2.3
